src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,451 @@
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import threading
+import uuid
+import re
+import time
+import os
+
+from general_utils import run_task
+from command_utils_base import CommandFailure
+from ior_test_base import IorTestBase
+from write_host_file import write_host_file
+from test_utils_pool import TestPool
+from ior_utils import IorCommand
+from job_manager_utils import Mpirun
+from mpio_utils import MpioUtils
+
+try:
+    # python 3.x
+    import queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+def get_device_ids(dmg, servers):
+    """Get the NVMe Device ID from servers
+
+    Args:
+        dmg: DmgCommand class instance.
+        servers (list): list of server hosts.
+
+    Returns:
+        devices (dictionary): Device UUID for servers.
+
+    """
+    devices = {}
+    dmg.set_sub_command("storage")
+    dmg.sub_command_class.set_sub_command("query")
+    dmg.sub_command_class.sub_command_class.set_sub_command("list-devices")
+    for host in servers:
+        dmg.hostlist = host
+        try:
+            result = dmg.run()
+        except CommandFailure as _error:
+            raise "dmg command failed for list-devices"
+        drive_list = []
+        for line in result.stdout.split('\n'):
+            if 'UUID' in line:
+                drive_list.append(line.split(':')[1])
+        devices[host] = drive_list
+    return devices
+
+class ServerFillUp(IorTestBase):
+    """
+    Class to fill up the servers based on pool percentage given.
+    It will get the drives listed in yaml file and find the maximum capacity of
+    the pool which will be created.
+    IOR block size will be calculated as part of function based on percentage
+    of pool needs to fill up.
+    """
+    # pylint: disable=too-many-ancestors
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, *args, **kwargs):
+        """Initialize a IorTestBase object."""
+        super(ServerFillUp, self).__init__(*args, **kwargs)
+        self.no_of_pools = 1
+        self.capacity = 1
+        self.no_of_servers = 1
+        self.no_of_drives = 1
+        self.pool = None
+        self.dmg = None
+        self.container_info = {}
+        self.set_faulty_device = False
+
+    def setUp(self):
+        """Set up each test case."""
+        # obtain separate logs
+        self.update_log_file_names()
+        # Start the servers and agents
+        super(ServerFillUp, self).setUp()
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
```

Comment:
You don't need to recreate the whole client host file. You can simply override the following param as follows: self.hostfile_clients_slots = None. By default it is 1.